websites/M/MangaHere/presence.ts

```diff
@@ -0,0 +1,144 @@
+    }
+    else if (pathname == "/latest/") {
+      data.details = "Browsing latest manga";
+      data.startTimestamp = browsingStamp;
+    }
+    else if (pathname == "/ranking/") {
+      data.details = "Browsing by ranking";
+      data.startTimestamp = browsingStamp;
+    }
+    else if (pathname == "/spoilers/") {
+      data.details = "Browsing spoilers and news";
+      data.startTimestamp = browsingStamp;
+    }
+    else if (pathname == "/directory/") {
+      data.details = "Browsing all manga";
+      data.startTimestamp = browsingStamp;
+    }
+    else if (pathname == "/on_going/") {
+      data.details = "Browsing ongoing manga";
+      data.startTimestamp = browsingStamp;
+    }
+    //ganre/new/
+    else if (pathname.endsWith("/new/")) {
+      const splitUrl = url.split('/');
+      if (splitUrl[1] == "new") {
+        data.details = "Browsing new manga";
+      } else {
+        data.details = "Browsing new " + splitUrl[1] + " manga";
+      }
+      data.startTimestamp = browsingStamp;
+    }
+    //ganre/completed/
+    else if (pathname.endsWith("/completed/")) {
+      let splitUrl = url.split('/');
+      if (splitUrl[1] == "completed") {
+        data.details = "Browsing completed manga";
+      } else {
+        data.details = "Browsing completed " + splitUrl[1] + " manga";
+      }
+      data.startTimestamp = browsingStamp;
+    }
+    //ganre/on_going/
+    else if (pathname.endsWith("/on_going/")) {
+      var url = pathname;
+      var splitUrl = url.split('/');
+      if (splitUrl[1] == "on_going") {
+        data.details = "Browsing ongoing manga";
+      } else {
+        data.details = "Browsing ongoing " + splitUrl[1] + " manga";
+      }
+      data.startTimestamp = browsingStamp;
+    }
+    
+
+    //Manga Viewing
+    else if (pathname.startsWith("/manga") && pathname.endsWith("/")) {
+      const title = document.querySelector(".detail-info-right-title-font").textContent;
+      let link = window.location.href;
+      data.details = "Viewing manga:";
+      data.state = title;
+      data.startTimestamp = browsingStamp;
+      data.buttons = [{ label: "Read", url: link }];
+      data.smallImageKey = "viewing";
+      data.smallImageText = "Viewing";
+    }
+    //Manga Reading
+    else if (pathname.startsWith("/manga") && pathname.endsWith(".html")) {
+      let title = document.querySelector(".reader-header-title-1").textContent;
+      let chapter = document.querySelector(".reader-header-title-2").textContent;
+      //setting up page progress
+      let current = document.querySelector('.pager-list-left span');
+      if(current == null){
+        data.state = chapter;
+      }else{
+        let len = current.children.length;
+        let totalPages = current.children[len - 2].textContent;
+        let readingPage = document.querySelector(".pager-list-left > span > .active").textContent;
+        let progress = readingPage + "/" + totalPages;
```

Comment:
merge lets?

Reply:
Sure, just got too used to php and stuff so getting used to ts again